demos/matrix_free/stokes.py.rst

```diff
@@ -0,0 +1,50 @@
+==================
+ Stokes Equations
+==================
+Again, they are what they are.  This is the driven cavity example.
+We have several options files showing how to do some field-splitting
+and such for this.::
+
+  from firedrake import *
+
+  N = 128
+
+  M = UnitSquareMesh(N, N)
+
+  V = VectorFunctionSpace(M, "CG", 2)
+  W = FunctionSpace(M, "CG", 1)
+  Z = V * W
+
+  u, p = TrialFunctions(Z)
+  v, q = TestFunctions(Z)
+
+  a = (inner(grad(u), grad(v)) -
+       p * div(v) +
+       div(u) * q + p*q)*dx
```

Comment:
That would be a Mistake.


On Mon, Aug 15, 2016 at 8:01 AM, Lawrence Mitchell <notifications@github.com
> wrote:

> In demos/matrix_free/stokes.py.rst
> <https://github.com/firedrakeproject/firedrake/pull/853#discussion_r74757545>
> :
>
> > +  from firedrake import *
> > +
> > +  N = 128
> > +
> > +  M = UnitSquareMesh(N, N)
> > +
> > +  V = VectorFunctionSpace(M, "CG", 2)
> > +  W = FunctionSpace(M, "CG", 1)
> > +  Z = V * W
> > +
> > +  u, p = TrialFunctions(Z)
> > +  v, q = TestFunctions(Z)
> > +
> > +  a = (inner(grad(u), grad(v)) -
> > +       p * div(v) +
> > +       div(u) * q + p*q)*dx
>
> Why is there a pressure mass term here @rckirby
> <https://github.com/rckirby>? I was expecting: inner(grad(u), grad(v))*dx
> + div(v)*p*dx + q*div(u)*dx.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/firedrakeproject/firedrake/pull/853/files/ff5ec4e0eabb94576ac69c9c82932eaa50b2d14e#r74757545>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADl6LzVDsmWQDAYWKnwh95fE9buV7sb5ks5qgGM5gaJpZM4JiV-g>
> .
>